ci: pyright is failing

### What happened?

```
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.3 (node_modules/pyright/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

+ pyright@1.1.355
added 1 package from 1 contributor and audited 2 packages in 1.224s
found 0 vulnerabilities

internal/modules/cjs/loader.js:626
    throw err;
    ^

Error: Cannot find module 'node:util'
Require stack:
- /home/runner/.cache/pyright-python/1.1.355/node_modules/pyright/dist/pyright.js
- /home/runner/.cache/pyright-python/1.1.355/node_modules/pyright/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:623:15)
    at Function.Module._load (internal/modules/cjs/loader.js:527:27)
    at Module.require (internal/modules/cjs/loader.js:681:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.9632 (/home/runner/.cache/pyright-python/1.1.355/node_modules/pyright/dist/pyright.js:1:610)
    at o (/home/runner/.cache/pyright-python/1.1.355/node_modules/pyright/dist/pyright.js:1:1243)
    at Object.1264 (/home/runner/.cache/pyright-python/1.1.355/node_modules/pyright/dist/vendor.js:2:794958)
    at o (/home/runner/.cache/pyright-python/1.1.355/node_modules/pyright/dist/pyright.js:1:1243)
    at Object.1476 (/home/runner/.cache/pyright-python/1.1.355/node_modules/pyright/dist/pyright-internal.js:1:442282)
    at o (/home/runner/.cache/pyright-python/1.1.355/node_modules/pyright/dist/pyright.js:1:1243) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/runner/.cache/pyright-python/1.1.355/node_modules/pyright/dist/pyright.js',
    '/home/runner/.cache/pyright-python/1.1.355/node_modules/pyright/index.js'
  ]
}
internal/modules/cjs/loader.js:626
    throw err;
    ^
```

### How can we reproduce the bug?

I think we need to update Node versions: https://github.com/microsoft/pyright/issues/7365

### Environment/Setup

_No response_

--- a/ci/conda_env_python.txt
+++ b/ci/conda_env_python.txt
@@ -17,6 +17,8 @@
 
 Cython
 importlib-resources
+# nodejs is required by pyright
+nodejs >=13.0.0
 pandas
 pyarrow>=8.0.0
 pyright

